cool-core/src/test/java/com/nus/cool/functionality/CsvLoaderTest.java

```diff
@@ -0,0 +1,74 @@
+package com.nus.cool.functionality;
+
+import com.nus.cool.core.util.config.CsvDataLoaderConfig;
+import com.nus.cool.core.util.config.DataLoaderConfig;
+import com.nus.cool.model.CoolLoader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+
+public class CsvLoaderTest {
+    static final Logger logger = LoggerFactory.getLogger(CsvLoaderTest.class);
+
+    @BeforeTest
+    public void setUp() {
+        logger.info("Start UnitTest " + CsvLoaderTest.class.getSimpleName());
+    }
+
+    @AfterTest
+    public void tearDown() {
+        logger.info(String.format("Tear down UnitTest %s\n", CsvLoaderTest.class.getSimpleName()));
+    }
+
+    @Test(dataProvider = "CsvLoaderTestDP")
```

Comment:
For this method which you can not prove it's true, you can add wrong test and check the exception.